packages/composites/token-allocation/src/dataProvider.ts

```diff
@@ -2,11 +2,40 @@ import { Requester } from '@chainlink/ea-bootstrap'
 import { RequestConfig } from '@chainlink/types'
 import { ResponsePayload } from './types'
 
-export const getPriceProvider = (apiConfig: RequestConfig, jobRunID: string) => async (
-  symbols: string[],
-  quote: string,
-  withMarketCap = false,
-): Promise<ResponsePayload> => {
+const batchingSupport: { [name: string]: boolean } = {
+  COINGECKO: true,
+  COINMARKETCAP: true,
+  CRYPTOCOMPARE: true,
+  NOMICS: true,
+}
+
+export const getPriceProvider = (
```

Comment:
How do you feel about documenting this function, since now it has grown in complexity and handles multiple use cases